tests/data_generator/datagenerator.go

```diff
@@ -0,0 +1,300 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"math/rand"
+	"strconv"
+
+	"github.com/spf13/viper"
+	"github.com/uc-cdis/cohort-middleware/config"
+	"github.com/uc-cdis/cohort-middleware/db"
+	"github.com/uc-cdis/cohort-middleware/tests"
+	"github.com/uc-cdis/cohort-middleware/utils"
+)
+
+type Concepts struct {
+	Concepts []Concept
+}
+
+type Concept struct {
+	Concept           string
+	Id                string
+	ConceptName       string   `mapstructure:"concept_name"`
+	ValueType         string   `mapstructure:"value_type"`
+	PersonIds         []int64  `mapstructure:"person_ids"`
+	PersonIdsOverride []int64  // we use this one for dynamic personIds from cohorts that are created on the fly
+	PossibleValues    []string `mapstructure:"possible_values"`
+	ConceptValueName  string   `mapstructure:"concept_value_name"`
+	RatioOfPersons    float32  `mapstructure:"ratio_of_persons"`
+	CloneCount        int      `mapstructure:"clone_count"`
+}
+
+type Cohorts struct {
+	Cohorts []Cohort
+}
+
+type Cohort struct {
+	Cohort          string
+	PersonIds       []int64 `mapstructure:"person_ids"`
+	NumberOfPersons int     `mapstructure:"number_of_persons"`
+	CloneCount      int     `mapstructure:"clone_count"`
+	Concepts        []Concept
+}
+
+func RunDataGeneration(testDataConfigFilePrefix string) {
+	conf := GetTestDataConfig(testDataConfigFilePrefix)
+
+	concepts := Concepts{}
+	conf.Unmarshal(&concepts)
+	AddConcepts(concepts.Concepts)
+
+	cohorts := Cohorts{}
+	conf.Unmarshal(&cohorts)
+	AddCohorts(cohorts.Cohorts)
+}
+
+func GetTestDataConfig(configFilePrefix string) *viper.Viper {
+	var err error
+	config := viper.New()
+	config.SetConfigType("yaml")
+	log.Printf("Reading test data config \"%s\"...", configFilePrefix)
+	config.SetConfigName(configFilePrefix)
+	config.AddConfigPath(".")
+	err = config.ReadInConfig()
+	if err != nil {
+		log.Fatal("error on parsing test data configuration file")
```

Comment:
Will this essentially handle validation of the schema of the YAML?

Reply:
No, it doesn't validate anything yet, other than that it is valid yaml. 
I now improved this part a bit to print out the error from the parser as well.

The solution as a whole does not validate the contents of the yaml yet. I'm not sure if we want to do this and how. I would like to implement any such validation in a future iteration, once we're happy with the results of this tool.